sktime/forecasting/compose/_multiplexer.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+from sktime.base import _HeterogenousMetaEstimator
+from sktime.forecasting.base._base import DEFAULT_ALPHA
+from sktime.forecasting.base._sktime import _OptionalForecastingHorizonMixin
+from sktime.forecasting.base._sktime import _SktimeForecaster
+from sklearn.base import clone
+
+__author__ = ["Kutay Koralturk"]
+__all__ = ["MultiplexForecaster"]
+
+
+class MultiplexForecaster(
+    _OptionalForecastingHorizonMixin, _SktimeForecaster, _HeterogenousMetaEstimator
+):
+    def __init__(
+        self,
+        forecasters: list,
+        selected_forecaster=None,
+    ):
+
+        """
+        MultiplexForecaster facilitates a framework for performing
+        model selection process over different model classes.
+        It should be used in conjunction with ForecastingGridSearchCV
+        to get full utilization.
+
+        Single use of MultiplexForecaster with forecasters
+        and selected_forecaster parameter specified,
+        works just like the selected component.
+        It does not provide any further use in that case.
+
+        When used with ForecastingGridSearchCV, MultiplexForecaster
+        provides an ability to compare different model class
+        performances with each other, just like a model tournament.
+        When ForecastingGridSearchCV is fitted with a MultiplexForecaster,
+        returned value for the selected_forecaster argument of best_params_
+        attribute of ForecastingGridSearchCV, gives the best
+        performing model class among given models provided in forecasters.
+
+        Parameters
+        ----------
+        forecasters : list
+            List of (forecaster names, forecaster objects)
+            MultiplexForecaster switches between these forecasters
+            objects when used with ForecastingGridSearchCV to
+            find the optimal model
+        selected_forecaster: str
+            An argument to make a selection among forecasters.
+            MultiplexForecaster uses selected_forecaster
+            to choose which component to fit.
+            Important for using with ForecastingGridSearchCV as a
+            hyperparameter.
+
+        Attributes
+        ----------
+        _forecaster : Sktime forecaster
+            forecaster that MultiplexForecaster will currently
+            forecast with.
+        _forecaster_fit_params: dict
+            Fit params for the forecaster that is
+            going to be used by the forecaster in the
+            _forecaster
+        """
+
+        self.forecasters = forecasters
+        self.selected_forecaster = selected_forecaster
+        self._check_components()
+
+        super(MultiplexForecaster, self).__init__()
+
+    def _check_components(self):
```

Comment:
Could we make use of the existing checks from [`_check_forecasters`](https://github.com/alan-turing-institute/sktime/blob/77f34fc09d1bfa6abac613048fe5407c1be0c387/sktime/forecasting/base/_meta.py#L29) and [`_check_names`](https://github.com/alan-turing-institute/sktime/blob/77f34fc09d1bfa6abac613048fe5407c1be0c387/sktime/base/_meta.py#L64) if possible?

Reply:
I decided to use _check_forecasters of `_HeterogenousEnsembleForecaster`